src/services/usersService.js

```diff
@@ -1,13 +1,132 @@
 import api from "@api/api";
+import showCustomAlert from "@utils/show_alert";
 import store from "@store/store";
 import { history } from "@helpers/history";
 
 class UsersService {
+  constructor() {
+    this.currentUser = store.getState().currentUser.value;
+  }
+
+  #showWarningAlert(message) {
+    showCustomAlert(message, "warning");
+    return true;
+  }
+
+  #validateFieldLength(value, min, max, fieldName) {
+    const length = value?.trim().length;
+    if (length < min || length > max) {
+      this.#showWarningAlert(
+        `The ${fieldName} length must be in the range from ${min} to ${max}.`
+      );
+      return true;
+    }
+    return false;
+  }
+
+  #validateEmail(email) {
+    if (email && !/^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/.test(email)) {
+      this.#showWarningAlert("The format of the email address is incorrect.");
+      return true;
+    }
+    return false;
+  }
+
+  #validatePhone(phone) {
+    if (phone && !/^[0-9]{3,15}$/.test(phone)) {
+      this.#showWarningAlert(
+        "The phone number should be 3 to 15 digits in length."
+      );
+      return true;
+    }
+    return false;
+  }
+
+  #validateIsEmptyObject(obj) {
+    const keys = Object.keys(obj);
+    if (!keys.length) {
+      return true;
+    }
+
+    let isValueEmpty = true;
+    keys.forEach((key) => !!obj[key]?.trim() && (isValueEmpty = false));
+
+    return isValueEmpty;
+  }
+
+  async sendEditRequest(data, typeOfValidation) {
+    if (this.#validateIsEmptyObject(data)) {
+      return;
+    }
+
+    switch (typeOfValidation) {
+      case "user":
+        if (
+          this.#validateFieldLength(data.first_name, 1, 20, "first name") ||
+          this.#validateFieldLength(data.last_name, 1, 20, "last name")
+        ) {
+          return false;
+        }
+        break;
+      case "personal":
+        if (
+          this.#validateEmail(data.email) ||
+          this.#validatePhone(data.phone)
+        ) {
+          return false;
+        }
+        break;
+      default:
+        break;
+    }
+
+    try {
+      return await api.userEdit(data);
+    } catch (err) {
+      showCustomAlert(err.message, "danger");
+      return false;
+    }
+  }
+
+  async sendSearchRequest(data) {
+    return await api.userSearch(data);
+  }
+
+  async changePasswordRequest() {
+    const currentPassword = window.prompt("Enter your current password:");
+    if (!currentPassword) {
+      return;
+    }
+
+    let newPassword = window.prompt("Enter a new password:");
```

Comment:
I'm still not sure whether it's a good idea or not to have this UI related things in a service